### PR TITLE
[WIP] Initial Sphinx setup

### LINKED ISSUE #14: Add sphinx documentation
## Feature Description
We need to create the first version of pyzpk lib docs.

The docs would be linked to a CI/CD testing environment to ensure that each new PRs to the stable branch does not break the documentation.

Here is the reference link for the [sphinx documentation](https://www.sphinx-doc.org/en/master/).

### PR DESCRIPTION
**Pull request by evalevanto**
## Description
Initial Sphinx documentation.
- [x] Sphinx setup.
- [x] Enable autodoc from Google format docstrings.
- [x] Added initial documentation: installing PyZPK.
**closes #14**
- [ ] Host on [readthedocs](https://docs.readthedocs.io/en/stable/intro/import-guide.html).

## Affected Dependencies
Sphinx==3.1.2

## How has this been tested?
To build documentation: `sphinx-build -b html ./source ./build` in docs dir.
- HTML documentation files generated

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: PyPi_deployment_steps.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Upload your package to PyPi
```
Comment (gargarchit) [inline PyPi_deployment_steps.txt:1]: We have this file already in dev branch, right?

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+# -- Project information -----------------------------------------------------
+
+project = "PyZPK"
+copyright = "2020, Benardi Nunes"
+author = "Benardi Nunes"
+
+# The full version, including alpha/beta/rc tags
+release = "0.0.1"
```
Comment (gargarchit) [inline docs/source/conf.py]: I think currently we can leave release as blank.

File: PyPi_deployment_steps.txt (hunk 1 above)
Comment (evalevanto) [reply PyPi_deployment_steps.txt:1]: Yes. I just fixed a typo in that file is all.

File: docs/source/conf.py (hunk 2 above)
Comment (evalevanto) [reply docs/source/conf.py:14]: Alright.
Also, the step for hosting the documentation on readthedocs, it is best to be done by maintainers for admin reasons.

File: PyPi_deployment_steps.txt (hunk 1 above)
Comment (gargarchit) [reply PyPi_deployment_steps.txt:1]: Oh Okay

**Comment by Benardi** [comment]
@evalevanto You might want to fix the authorship of your commits to ensure your contribution is computed.

**Comment by Benardi** [comment]
I've build the Sphinx documentation, all checks out. Good work!

image

However, no module documentation has been picked. 
* If it's a lack of docstrings you could a single one as a placeholder just to validate your documentation code.
* If Sphinx alone can't deal with our particular case (a Python wrap of C++), it might be necessary to resort to [doxygen](https://www.doxygen.nl/index.html) +[breathe](https://breathe.readthedocs.io/en/latest/) +  sphinx

File: PyPi_deployment_steps.txt (hunk 1 above)
Comment (chinmayshah99) [inline PyPi_deployment_steps.txt:1]: It won't be a good idea to give these instructions out here since we don't want multiple versions of PyZPK out.

**Comment by chinmayshah99** [comment]
> * If it's a lack of docstrings you could a single one as a placeholder just to validate your documentation code.

We can add docstrings in the pybind11 code. Example here: https://github.com/OpenMined/PyDP/blob/dev/src/bindings/PyDP/algorithms/distributions.cpp#L16

**Pull request closed without merging**